nextcord/http.py

```diff
@@ -268,152 +663,250 @@ async def request(
+        for retry_count in range(max_retry_count):  # To prevent infinite loops.
+            should_retry = False
+            try:
+                async with global_rate_limit:
+                    async with url_rate_limit:
+                        # This check is for asyncio.gather()'d requests where the rate limit can change.
+                        if (
+                            temp := self._get_url_rate_limit(route.method, route, auth)
+                        ) is not url_rate_limit and not None:
+                            temp = cast(RateLimit, temp)
                             _log.debug(
-                                "A rate limit bucket has been exhausted (bucket: %s, retry: %s).",
-                                bucket,
-                                delta,
+                                "Route %s had the rate limit changed, resetting and retrying.",
+                                rate_limit_path,
                             )
-                            self._dispatch(
-                                "http_ratelimit",
-                                int(limit),
-                                int(remaining),
-                                delta,
-                                bucket,
-                                response.headers.get("X-RateLimit-Scope"),
+                            url_rate_limit = temp
+                            continue
+
+                        if files:
+                            for f in files:
+                                f.reset(seek=retry_count)
+
+                        if form:
+                            form_data = aiohttp.FormData(quote_fields=False)
+                            for params in form:
+                                form_data.add_field(**params)
+                            kwargs["data"] = form_data
+
+                        async with self.__session.request(
+                            method=route.method,
+                            url=route.url,
+                            headers=headers,
+                            proxy=self._proxy,
+                            proxy_auth=self._proxy_auth,
+                            **kwargs,
+                        ) as response:
+                            _log.debug(
+                                "%s %s with %s has returned %s",
+                                route.method,
+                                route.url,
+                                kwargs.get("data"),
+                                response.status,
                             )
-                            maybe_lock.defer()
-                            self.loop.call_later(delta, lock.release)
-
-                        # the request was successful so just return the text/json
-                        if 300 > response.status >= 200:
-                            _log.debug("%s %s has received %s", method, url, data)
-                            return data
```

Comment:
`should_retry` is `False` by default. If no errors are raised and it's never set to `True`, the if statement checking `should_retry` will break execution out of the loop and it will hit the `return` at the end of the function. Different styles of coding, I suppose.

Reply:
Oh, I missed that part of the code. My bad.